test/Microsoft.AspNetCore.Routing.Tests/Internal/UriBuildingContextTest.cs

```diff
@@ -28,66 +30,63 @@ public void EncodeValue_EncodesEntireValue_WhenEncodeSlashes_IsFalse()
 
             // Assert
             Assert.Equal(expected, uriBuilldingContext.ToString());
         }
 
         [Theory]
-        [InlineData("a/b b1/c", 0, 2, "/UrlEncode[[a]]/")]
-        [InlineData("a/b b1/c", 3, 4, "/UrlEncode[[ b1]]/")]
-        [InlineData("a/b b1/c", 3, 5, "/UrlEncode[[ b1]]/UrlEncode[[c]]")]
+        [InlineData("a/b b1/c", 0, 2, "/a/")]
+        [InlineData("a/b b1/c", 3, 4, "/%20b1/")]
+        [InlineData("a/b b1/c", 3, 5, "/%20b1/c")]
         [InlineData("a/b b1/c/", 8, 1, "/")]
         [InlineData("/", 0, 1, "/")]
-        [InlineData("/a", 0, 2, "/UrlEncode[[a]]")]
-        [InlineData("a", 0, 1, "/UrlEncode[[a]]")]
-        [InlineData("a/", 0, 2, "/UrlEncode[[a]]/")]
+        [InlineData("/a", 0, 2, "/a")]
+        [InlineData("a", 0, 1, "/a")]
+        [InlineData("a/", 0, 2, "/a/")]
         public void EncodeValue_EncodesOnlySlashes_WithinSubsegment_WhenEncodeSlashes_IsFalse(
             string value,
             int startIndex,
             int characterCount,
             string expected)
         {
             // Arrange
-            var urlTestEncoder = new UrlTestEncoder();
-            var uriBuilldingContext = new UriBuildingContext(urlTestEncoder);
+            var uriBuilldingContext = new UriBuildingContext(UrlEncoder.Default);
 
             // Act
-            uriBuilldingContext.EncodeValue(value, startIndex, characterCount, encodeSlashes: false);
+            uriBuilldingContext.EncodePathValue(value, startIndex, characterCount, encodeSlashes: false);
 
             // Assert
             Assert.Equal(expected, uriBuilldingContext.ToString());
         }
-
+        
         [Theory]
-        [InlineData("/Author", false, false, "/UrlEncode[[Author]]")]
-        [InlineData("/Author", false, true, "/UrlEncode[[Author]]")]
-        [InlineData("/Author", true, false, "/UrlEncode[[Author]]/")]
-        [InlineData("/Author", true, true, "/UrlEncode[[Author]]/")]
-        [InlineData("/Author/", false, false, "/UrlEncode[[Author]]/")]
-        [InlineData("/Author/", false, true, "/UrlEncode[[Author/]]")]
-        [InlineData("/Author/", true, false, "/UrlEncode[[Author]]/")]
-        [InlineData("/Author/", true, true, "/UrlEncode[[Author/]]/")]
-        [InlineData("Author", false, false, "/UrlEncode[[Author]]")]
-        [InlineData("Author", false, true, "/UrlEncode[[Author]]")]
-        [InlineData("Author", true, false, "/UrlEncode[[Author]]/")]
-        [InlineData("Author", true, true, "/UrlEncode[[Author]]/")]
+        [InlineData("/Author", false, false, "/Author")]
+        [InlineData("/Author", false, true, "/Author")]
+        [InlineData("/Author", true, false, "/Author/")]
+        [InlineData("/Author", true, true, "/Author/")]
+        [InlineData("/Author/", false, false, "/Author/")]
+        [InlineData("/Author/", false, true, "/Author%2F")]
+        [InlineData("/Author/", true, false, "/Author/")]
+        [InlineData("/Author/", true, true, "/Author%2F/")]
```

Comment:
Are these expected paths right?

Reply:
Yeah this is kinda a janky test - it's taking a single string/segment, then encoding it, then getting the result as a full path. 

It's doing a cross product of all possible cases instead of factoring the code into independent testable units. Theory tests are always a bad decision.